### ISSUE TITLE
broken firefox 34.0 and Xubuntu 14.04.1 LTS

### ISSUE THREAD

**Issue by foudfou** [foudfou/FireTray#147]
[reported by Jochen]

I am using fire tray to get a nice icon displaying
the number of unread messages.
On clicking on that icon, a thunderbird main window
opens - so everything worked like a charm.
Till today.
There seems to be an update of systemtray
(Xubuntu had no update since several days)
which distroyed everyting - no number of unread
mails, no function when clicking on the item
(which is diaplayed as the standard bird now -
 not the letter or the card with the number).
Will this bug get fixed soon or is a revert to
the former functionality possible?
Many thanks in advance for your help!

P.S.: If this problem will stay - do you know of
  any other solution to get an icon displaying the
  number of unread mails?
P.P.S.: I am using latest firefox (34.0) and updated
  Xubuntu 14.04.1 LTS, `Trusty Tahr'.
  I used: `Mail notification type'
               -  `display new message count'

**Comment by foudfou** [comment]
Hi, thanks for reporting!
Few things to check before we go further:
* any FireTray-related messages in the JS/Error-console ?
* does the problem occur with a different profile (`thunderbird -P`) ?
* with what version did the problem occur ? re-install previous versions from [AMO](https://addons.mozilla.org/en-US/firefox/addon/firetray/versions/)

**Comment by RobinHerink** [comment]
I signed-up with GitHub to report the same thing.

I cannot change the icon in sys-tray (the default blue bird), but changing the "new mail" icon works.
Left-clicking the icon does nothing, only middle-click works and even then the window doesn't come on top.
Sometimes no icon appears and I have to reinstall FireTray.

Going back to 0.5.3 solved all problems.

My system is Linux Mint 17 KDE.

**Comment by foudfou** [comment]
I suspect FireTray is using libappindicator, when you may not want it to do so. 

Can you issue:
```
dbus-send --session --dest=org.freedesktop.DBus --type=method_call \
  --print-reply /org/freedesktop/DBus org.freedesktop.DBus.ListNames \
  | grep org.kde.StatusNotifierWatcher
```

If it returns something, then you can try to unset the `with_appindicator` option in TB's config editor, and restart TB. Tell us how it goes.

**Comment by RobinHerink** [comment]
Yes, that command outputs:
string "org.kde.StatusNotifierWatcher"

Something strange is happening after that appindicator change.
Now Thunderbird doesn't show any icon in systray. Only when I launch it with the terminal it gives this error and then launches: 

(process:8079): GLib-CRITICAL **: g_slice_set_config: assertion 'sys_page_size == 0' failed
[calBackendLoader] Using libical backend at /home/robin/.thunderbird/y9qbzlsf.default/extensions/{e2fda1a4-762b-4020-b5ad-a41df1933103}/components/libical.manifest

**Comment by RobinHerink** [comment]
Back to 0.5.3 : 
Starting in terminal gives the same "error" so it probably isn't an error.
So I don't understand why 0.5.4 only works from the terminal and not from a shortcut!

**Comment by foudfou** [comment]
@RobinHerink have you tried unsetting `with_appindicator` ?

**Comment by RobinHerink** [comment]
If you mean extensions.firetray.with_appindicator  then yes, that's what I wrote about: "something strange is happening after that appindicator change"

**Comment by KabbAmine** [comment]
@RobinHerink The error ...

```
(process:8079): GLib-CRITICAL **: g_slice_set_config: assertion 'sys_page_size == 0' failed
[calBackendLoader] Using libical backend at /home/robin/.thunderbird/y9qbzlsf.default/extensions/{e2fda1a4-762b-4020-b5ad-a41df1933103}/components/libical.manifest
```

... is related to lightning.

**Comment by JPRuehmann** [comment]
I have this problem to, But I dont understand anything of the above.
Can anyone tell me how I can Fix it in Simple Terms.

Thamks,

**Comment by foudfou** [comment]
@JPRuehmann take a look at [the background](/foudfou/FireTray/wiki/AppIndicators-StatusNotifierItem).
For the fix, try Preferences>Advanced>Config Editor
search `extensions.firetray.with_appindicator`, set it to false, and restart Thunderbird.

**Comment by RobinHerink** [comment]
@KabbAmine Alright, after disabling Lightning the "error" is gone but it has no effect on the problem discussed here. Thunderbird doesn't show the systray icon any other way than starting it from the terminal.

@JPRuehmann Right now the only solution seems to be to go back to v. 0.5.3.

**Comment by vaskur** [comment]
I'm also affected by this bug. Also Xubuntu 14.04 LTS,  Thunderbird 31.3.0. The workaround described by foudfou works though.

**Comment by KabbAmine** [comment]
@RobinHerink Of course, its not lightning's fault :smile: Looks like your problem is more related to KDE, I don't use it, but check the options of the system tray widget in your panel, maybe thunderbird icon is set as hidden (Or check if you can add another systray widget).

**Comment by sgrayban** [comment]
I unset with_appindicator in about:config and set it to false and all is ok now... I don't think we should keep it like that as other addons do rely on that function.

**Comment by RobinHerink** [comment]
@KabbAmine Yep, it seems related to KDE. But since I have no problems with other apps using the systray and 0.5.3 works as expected, the problem must be something introduced in 0.5.4.

**Comment by foudfou** [comment]
@sgrayban `with_appindicator` is specific to FireTray.

**Comment by boussouira** [comment]
Problem fixed by setting with_appindicator to false (Kubuntu 14.04.1)

**Comment by konbeto** [comment]
I also can confirm this bug. I'm using KDE, Kubuntu 14.04. As a workaround, I downgraded to 0.5.3.

**Comment by patzi** [comment]
I had the same problem using Xubuntu 14.04. Setting **extensions.firetray.with_appindicator** to **false** in the config editor of thunderbird worked for me.

**Comment by jombra** [comment]
Sorry for the delay.
I checked with my friend and he experienced the same bug (on Mint with XFCE) -
and found the solution (or at least workaround) for this new year problem - which is
caused by the new version of FireTray (0.5.4):
~
- In Thunderbird:
  `Edit' - Preferences - `Advanced' (cogwheel symbol) - `General' (Tab)
         - `Config Editor' (OK to warning: I'll be careful, I promise)
- In list search for "extensions.firetray.with_appindicator" and
  double click (changing value from `true' to `false')
- Restart Thunderbird
- Thunderbird-Icon - Left Click - Preferences (belonging to FireTray)
- Enable mail notification - message count type: x unread messages
- Mail notification type:
  display new message count, Text color: <your favorite color>
~
Thus, everything is absolutely the same with the new version as it was
before ...
I hope this will help others, too - and may also help to fix the problem/avoid
similar ones in future.
JMB

**Comment by jcvernaleo** [comment]
@jombra that solution also fixed my issue with both firefox and thunderbird #155 

Thanks!

**Comment by gerstl** [comment]
Setting with_appindicator to false brings back the tray icon's functionality, but it doesn't solve the problem that the app (Thunderbird - well, icedove in my case, running KDE under Debian) does not come up when launched from the KDE menu or panel icon. Instead I get a defunct process that I need to kill:
   17068  5216 38 17:44 ?        00:00:01 /usr/bin/icedove
   17069 17068  0 17:44 ?        00:00:00 [icedove] <defunct>

Only downgrading to FireTray 0.5.3 solves the problem, i.e. everything works fine then.

**Comment by RobinHerink** [comment]
@gerstl Thank you, I thought I was the only one experiencing this. 
(Linux Mint KDE 17.1)

**Comment by ntassani** [comment]
Hi! I read all the discussion, and try everithing, but setting extensions.firetray.with_appindicator to false dont work in elementary luna. Because the indicator stops to show on the panel.
What can i do? Thanks for your help.

**Comment by YamashitaRen** [comment]
You can't do anything.
Appindicator does't support the "left click".
So either you enable libappindicator support and the indicator will show in elementary, or either you use a desktop that support the old systray behavior.

More info : https://github.com/syncthing/syncthing-gtk/issues/73#issuecomment-74874981

**Comment by Lastique** [comment]
I have this problem in KDE5. Setting `extensions.firetray.with_appindicator` to false makes Thunderbird crash on the next start, and on the second try it starts but there is no icon in the system tray.

**Comment by YamashitaRen** [comment]
Currently, you have to keep appindicator support set to true with Plasma5 too.
http://blog.martin-graesslin.com/blog/2014/03/system-tray-in-plasma-next/

**Comment by anamilodesu** [comment]
So does this issue has a solution for I encountered the same problem as the previous commentator
"Setting extensions.firetray.with appindicator to false makes Thunderbird crash on the next start, and on the second try it starts but there is no icon in the system tray."
Only I use ubuntu 15.10 with unity

**Comment by dmatej** [comment]
Kubuntu 15.10 - any configuration does not work, icon does not show, only "window draws message on new messages" works.
